src/storage/src/hummock/compactor.rs

```diff
@@ -126,6 +131,54 @@ impl CompactionFilter for StateCleanUpCompactionFilter {
     }
 }
 
+#[derive(Clone)]
+pub struct TTLCompactionFilter {
+    table_id_to_ttl: HashMap<u32, u32>,
+    expire: u64,
+}
+
+impl CompactionFilter for TTLCompactionFilter {
+    fn filter(&self, key: &[u8]) -> bool {
+        let table_id = get_table_id(key);
+        match table_id {
+            Some(table_id) => {
+                let ttl = self.table_id_to_ttl[&table_id];
+                let epoch = get_epoch(key);
```

Comment:
Get epoch and table_id in once call for performance

Reply:
shall we need a new function to get epoch and table_id in once call ? (now not exist)

```rust
fn extract_table_id_and_epoch(full_key: &[u8]) -> (Option<u32>, Epoch)
```

Reply:
yes. A new function.

Reply:
fix it